lib/src/neu/verifications/contract-mismatcher.ts

```diff
@@ -328,6 +344,76 @@ export class ContractMismatcher {
     );
   }
 
+  private getQueryParamsArraySerializationStrategy(): { comma: boolean } {
+    const comma =
+      this.contract.config.paramSerializationStrategy.query.array === "comma";
+
+    return { comma };
+  }
+
+  private findMismatchOnRequestQueryParams(
+    endpoint: Endpoint,
+    userInputRequest: UserInputRequest
+  ): Result<Mismatch[], Error> {
+    const queryParamsString = url.parse(userInputRequest.path).query || "";
+    const contractQueryParams = endpoint.request!!.queryParams;
+
+    const queryParams = qs.parse(queryParamsString, {
+      ...this.getQueryParamsArraySerializationStrategy()
+    });
+
+    let result;
+    let mismatches: Mismatch[] = [];
+    for (const {
+      name: queryParamName,
+      optional,
+      type: contractQueryParamType
+    } of contractQueryParams) {
+      const requestQueryParam = queryParams[queryParamName];
+
+      // Query parameter is mandatory and hasn't been provided
+      if (typeof requestQueryParam === "undefined" && !optional) {
```

Comment:
```suggestion
      if (requestQueryParam === undefined && !optional) {
```

Reply:
I'd stick to using "typeof",  as
1. Strict equality operator can go wrong if the variable is undeclared
2. undefined could be shadowed to maybe redefined.

"typeof" will always work, it could never go wrong